### PR TITLE
Added changes for attaching data volumes to store NBDE data

### PR DESCRIPTION
**Pull request by aditijadhav38**
- Upgrade Ansible Galaxy versions
- Add logging to each part of the ansible playbook calls
- Add support for Data Disk creation
- Minor changes in main.tf for fetching bastion_public_ip

### CODE REVIEW

File: modules/2_nbde/files/powervs-setup.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@
           con_name=$(nmcli -t -f NAME connection show | grep $env)
           nmcli connection modify "$con_name" ethtool.feature-rx off
           nmcli connection modify "$con_name" ethernet.mtu {{private_network_mtu}}
+          nmcli connection modify "$con_name" ipv4.dns {{ipv4_dns}}
```
Comment (prb112) [inline modules/2_nbde/files/powervs-setup.yml:23]: Thanks for fixing this dns issue

File: modules/2_nbde/files/powervs-setup.yml [hunk 2]
```diff
@@ -82,6 +83,7 @@
           # Setup RHEL9 subscription
           os_ver=$(cat /etc/os-release | egrep "^VERSION_ID=" | awk -F'"' '{print $2}')
           if [[ $os_ver != "9"* ]]; then
+            sudo yum update -y
```
Comment (prb112) [inline modules/2_nbde/files/powervs-setup.yml:86]: What's the reason for the `yum update -y` here? 

sudo yum update -y at L86 is duplicate of https://github.com/IBM/powervs-tang-server-automation/blob/a941cae3fd8b8dc61176a73106dd9582f197266d/modules/2_nbde/files/powervs-setup.yml#L96

File: modules/2_nbde/nbde.tf [hunk 3]
```diff
@@ -159,6 +205,13 @@ EOF
...
     inline = [
       <<EOF
+cat << EOT > ansible.cfg
+[defaults]
+retry_files_enabled = False
+pipelining          = True
+host_key_checking   = False
+log_path            = /root/tang-remove-subscription-logs.txt
+EOT
 ANSIBLE_HOST_KEY_CHECKING=False ansible-playbook -i inventory powervs-remove-subscription.yml
```
Comment (prb112) [inline modules/2_nbde/nbde.tf:215]:
```suggestion
ANSIBLE_CONFIG=ansible.cfg ansible-playbook -i inventory powervs-remove-subscription.yml
```

File: modules/2_nbde/files/powervs-setup.yml (hunk 1 above)
Comment (prb112) [inline modules/2_nbde/files/powervs-setup.yml:23]:
```suggestion
          nmcli connection modify "$con_name" ipv4.dns "{{ipv4_dns}}"
```

Comment (prb112) [reply modules/2_nbde/files/powervs-setup.yml:23]: Let's defensively put this in a double quote.

File: modules/3_fips/files/fips.yml [hunk 4]
```diff
@@ -6,4 +6,10 @@
...
       shell: |
-        fips-mode-setup --enable
\ No newline at end of file
+        fips-mode-setup --enable
++    - name: Reboot the host
++      become_user: root
++      ansible.builtin.reboot:
++        reboot_timeout: 3000
++        pre_reboot_delay: 1000
++        post_reboot_delay: 2000
```
Comment (prb112) [inline modules/3_fips/files/fips.yml:15]:
```suggestion
    - name: Reboot the tang servers
      become_user: root
      ansible.builtin.reboot:
        reboot_timeout: 3000
        pre_reboot_delay: 1000
        post_reboot_delay: 2000
```

File: modules/3_fips/fips.tf [hunk 5]
```diff
@@ -101,24 +104,8 @@ resource "null_resource" "bastion_fips_enable" {
       <<EOF
 # enable FIPS as required
 sudo fips-mode-setup --enable
+sudo shutdown -r +1
```
Comment (prb112) [inline modules/3_fips/fips.tf:107]:
```suggestion
# We considered using ibm_pi_instance_action and opted for an Operating Systems controlled reboot as we know the state when we take this action.
sudo shutdown -r +1
```

File: variables.tf [hunk 6]
```diff
@@ -69,14 +67,20 @@ variable "bastion_public_ip" {
...
   # only three nodes are supported
   default = {
-    count      = 3
-    memory     = "16"
-    processors = "1"
+    count             = 3
+    memory            = "16"
+    processors        = "1"
+    data_volume_size  = 10 #Default volume size (in GB) to be attached to the tang server nodes.
+    data_volume_count = 1  #Number of volumes to be attached to each tang server node.
```
Comment (prb112) [inline variables.tf:74]:
```suggestion
    data_volume_size  = 10 # Default volume size (in GB) to be attached to the tang server nodes.
    data_volume_count = 1  # Number of volumes to be attached to each tang server node.
```

**Review by prb112 — CHANGES_REQUESTED** [review]
Per comments left in related issue.

File: modules/2_nbde/files/powervs-setup.yml (hunk 2 above)
Comment (aditijadhav38) [reply modules/2_nbde/files/powervs-setup.yml]: Removed the line because its duplicate.

**Pull request merged**